src/som/primitives/actors/PromisePrims.java

```diff
@@ -88,11 +94,43 @@ public final SImmutableObject createPromisePair(final Object nil,
 
     @Override
     protected boolean isTaggedWithIgnoringEagerness(final Class<?> tag) {
-      if (tag == CreatePromisePair.class || tag == ExpressionBreakpoint.class) {
+      if (tag == CreatePromisePair.class || tag == ExpressionBreakpoint.class || tag == StatementTag.class) {
         return true;
       }
       return super.isTaggedWithIgnoringEagerness(tag);
     }
+
+    private boolean hasPromiseResolutionBreakpoint() {
```

Comment:
Because I think I need to enable the stepping for the primitives `createPromisePrim`, `whenResolved`, `whenResolvedOnError` and `onError`. And I don't see a common superclass for all of them where to put these two methods.